Competitive programming: Activity Selction Problem

Jinal4502 opened the pull request:
<hr>

## Description 📜

I have explained the activity selection algorithm alongwith its implementation in Python.
Fixes #7802 

<hr>

## Type of change 📝

<!----Please delete the hashtag from the correct option----->

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

<!----Please delete the hashtag from your domain----->

- [x] Competitive Programming
<hr>

## Checklist ✅

<!----Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this----->

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<!----Please delete options that are not relevant.----->

## Screenshots / Gif (Optional) 📸

<hr>

Jinal4502 commented:
Hey, I don't know why, but it is also showing me the old commits in the current pr which are already merged.

Jinal4502 commented:
I tried to remove those old commits by changing branch to other and again setting it back, but it didn't work. Anyway, even the old commits of my last pr https://github.com/girlscript/winter-of-contributing/pull/7657, are remerged it won't be a problem as they are already approved once. So, please review my pr.

ravireddy07 reviewed (commented):
Only One file is allowed per contribution. So, Add one file only.
And seems like you have deleted some file(`Competitive_Programming/Python/Greedy/README.md`), Please add it. Only add your file only.

ravireddy07 commented:
> #7657

But with-respect-to-PR, we need to maintain that clarity that in which PR what exactly got merged.
So, the previous PR got merged. you can update your branch and remove unrequired files.

Jinal4502 commented:
Hey, I updated my forked repository and now only 2 files are there in the 'files changed' section, one is my own file and second is the one I deleted, so I added it back.

ravireddy07 commented on Competitive_Programming/Python/Greedy/activity_selection.md:
````diff
@@ -0,0 +1,107 @@
+
+Here, we will use the <b>greedy approach</b> to address this problem.
+
+```
+Greedy Algorithm - It is an algorithmic strategy that makes the best optimal choice at each small stage which will eventually lead to an optimum solution for the entire problem. It always picks the best immediate output unlike dynamic approach which takes all the test cases into consideration.
+```
````
Better you just highlight this sentance. The reader can't scroll to right till end, it's a bad user experience.

ravireddy07 commented on Competitive_Programming/Python/Greedy/activity_selection.md:
```diff
@@ -0,0 +1,107 @@
+
+<img src='https://user-images.githubusercontent.com/65852362/141511389-d8de0ade-398f-45eb-9356-9698b68f3b04.png'>
+
+While choosing activities, we need to keep in mind that two activities i and j are compatible if the half-open interval <b>[s<sub>i</sub> >= f<sub>j</sub>)</b> or <b>[s<sub>j</sub> >= f<sub>i</sub>)</b>.
```
Can you add a note about what is **_s<sub>i</sub>, f<sub>j</sub>_**...etc
It will be easy to read.

ravireddy07 commented on Competitive_Programming/Python/Greedy/activity_selection.md:
````diff
@@ -0,0 +1,107 @@
+
+## Applications:
+
+```
+1. It is used for the scheduling multiple competing events in a room.
+2. It can also be used to schedule the manufacturing of multiple products on the same machine, where each product has its own production timelines.
+3. It is one of the most well-known generic problems used in Operations Research for dealing with real-life business problems.
+```
````
same for this, just highlight it

ravireddy07 reviewed (changes requested):
After going through the doc, As a reader i've few concerns like

* What about other resources like Q,R,T,U,V..etc, they'll be picked or not..?
* Is it always starts from first resource or which has minimum starting time..?
* How does the complexity defines for this

can you explain this points in the doc so that the reader can exit without doubts. Please add more points few you came across

Jinal4502 commented:
@ravireddy07, I have added explanation for not selecting activities Q, R, T, U, V, X and Y. Also, I have added the reason for choosing the first activity alongwith a brief explanation on time complexity. Also, I have removed greedy-definition and applications from the code block. And lastly, I have also added a short note about  si, fi, sj and fj. You can review all the changes, thank you!

ravireddy07 reviewed (changes requested):
Good work @Jinal4502 ,
You can remove/unstag this file(`Competitive_Programming/Python/Greedy/README.md`)

Jinal4502 commented:
@ravireddy07, okay, I removed it.
